src/expr/src/vector_op/array_min.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::{array::*, types::{Scalar, DefaultOrdered, ScalarImpl}, types::ToOwnedDatum};
+use risingwave_expr_macro::function;
+
+use crate::{Result, ExprError};
+
+#[function("array_min(list) -> int32")] 
+pub fn array_min(list: ListRef<'_>) -> Result<i32> {
```

Comment:
I guess the only way we can get it work with the current macro is:

```suggestion
#[function("array_min(list) -> int16")]
#[function("array_min(list) -> int32")] 
#[function("array_min(list) -> int64")]
// ... enumerate all capable types
// or #[function("array_min(list) -> *")]
// if all types are capable
pub fn array_min<T: Scalar>(list: ListRef<'_>) -> Result<T> {
```

just like what we did in `array_access`.

You may have a try and see if it works. 😇

Reply:
I've actually tried some similar signatures as this earlier, just by referencing `array_access`.
But the conversion between the input & output seems a bit tricky to get it to work, will try this again tomorrow🥺

Reply:
Okay. Thanks!

Reply:
I've modify the code in this way and it seems work.

Reply:
Yes, I once designed the `any` type which corresponds to `ScalarImpl` in Rust. But it needs some modification in the macro to support. We will consider adding `any` in the future.

Reply:
Sure, at that moment this could be refactored.